core/src/main/java/org/mule/api/debug/FieldDebugInfoFactory.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.api.debug;
+
+import org.mule.api.MuleEvent;
+import org.mule.util.AttributeEvaluator;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Creates instances of  {@link FieldDebugInfo}
+ *
+ * @since 3.8.0
+ */
+public class FieldDebugInfoFactory
+{
+
+    private FieldDebugInfoFactory()
+    {
+    }
+
+    private static final Map<Class, AttributeEvaluationExecutor> attributeEvaluationExecutors = new HashMap<>();
+    private static final AttributeEvaluationExecutor defaultAttributeEvaluationExecutor = new ObjectAttributeEvaluatorExecutor();
+
+    static
+    {
+        attributeEvaluationExecutors.put(String.class, new StringAttributeEvaluatorExecutor());
+        attributeEvaluationExecutors.put(Boolean.class, new BooleanAttributeEvaluatorExecutor());
+        attributeEvaluationExecutors.put(Integer.class, new IntegerAttributeEvaluatorExecutor());
+        attributeEvaluationExecutors.put(Object.class, defaultAttributeEvaluationExecutor);
+    }
+
+    private interface AttributeEvaluationExecutor
+    {
+        Object evaluate(MuleEvent event, AttributeEvaluator attributeEvaluator);
+    }
+
+    private static class BooleanAttributeEvaluatorExecutor implements AttributeEvaluationExecutor
```

Comment:
Is the difference in naming deliberate or accidental? 'Evaluator' vs. 'Evaluation'.